src/main/java/net/machpi/runelite/influxdb/MeasurementCreator.java

```diff
@@ -207,4 +211,31 @@ public Series createKillCountSeries(String boss) {
     }
 
     private static final int THRESHOLD = 50_000;
+
+    public Optional<Series> createActivitySeries(ActivityState.State lastState) {
+        Series.SeriesBuilder seriesB = createSeries().measurement(SERIES_ACTIVITY);
+        if (!Strings.isNullOrEmpty(lastState.getSkill())) {
+            seriesB = seriesB.tag("skill", lastState.getSkill());
+        }
+        if (!Strings.isNullOrEmpty(lastState.getLocationType())) {
+            seriesB = seriesB.tag("type", lastState.getLocationType());
+        }
+        if (!Strings.isNullOrEmpty(lastState.getLocation())) {
+            seriesB = seriesB.tag("location", lastState.getLocation());
+        }
+
+        Series series = seriesB.build();
+        if (Strings.isNullOrEmpty(series.getTags().getOrDefault("user", null)))
+            return Optional.empty();
+
+        return Optional.of(series);
+    }
+
+    public Optional<Measurement> createActivityMeasurement(ActivityState.State lastState) {
+        Optional<Series> series = createActivitySeries(lastState);
+        return series.map(value -> Measurement.builder()
+                .series(value)
+                .numericValue("t", Instant.now().getEpochSecond())
```

Comment:
this is probably a hack to avoid the built in deduplication of the writer system, but it should be possible to just write a different filter for this series.  You probably actually want to create a terminal op that enforces your desired spacing (or limits the spacing if you stop storing the current activity as a tag)